src/com/google/javascript/jscomp/deps/ModuleLoader.java

```diff
@@ -149,32 +157,80 @@ public String toModuleName() {
     }
 
     /**
-     * Find a CommonJS module {@code requireName}. See
+     * Find a JS module {@code requireName}. See
+     * https://nodejs.org/api/modules.html#modules_all_together
+     *
+     * @return The normalized module URI, or {@code null} if not found.
+     */
+    @Nullable
+    public ModulePath resolveJsModule(String moduleAddress) {
+      return resolveJsModule(moduleAddress, null, -1, -1);
+    }
+
+    /**
+     * Find a JS module {@code requireName}. See
      * https://nodejs.org/api/modules.html#modules_all_together
      *
      * @return The normalized module URI, or {@code null} if not found.
      */
-    public ModulePath resolveCommonJsModule(String requireName) {
-      String loadAddress;
+    @Nullable
+    public ModulePath resolveJsModule(
+        String moduleAddress, String sourcename, int lineno, int colno) {
+      String loadAddress = null;
 
       // * the immediate name require'd
-      if (isAbsoluteIdentifier(requireName) || isRelativeIdentifier(requireName)) {
-        loadAddress = resolveCommonJsModuleFileOrDirectory(requireName);
-      } else {
-        loadAddress = resolveCommonJsModuleFromRegistry(requireName);
+      switch (resolutionMode) {
+        case LEGACY:
+          loadAddress = resolveJsModuleFile(moduleAddress);
```

Comment:
Shiny!
I'd really like to see the PR show a clear preservation of the previous behavior for the LEGACY code path, though, rather than just a fix for this one case.

Could you do something like this?

1. Define private `ModuleLoader.legacyResolveJsModule()` (& related methods) that completely preserves current behavior in a way that is clear in the diff.
2. Call that from the `LEGACY` case in the switch statement.

Thanks.